app/components/avo/fields/common/badge_viewer_component.rb

```diff
@@ -1,9 +1,9 @@
 # frozen_string_literal: true
 
 class Avo::Fields::Common::BadgeViewerComponent < Avo::BaseComponent
-  def initialize(value:, options:)
-    @value = value
-    @options = options
+  prop :value, String
+  prop :options, Hash
+  prop :backgrounds, _Nilable(Hash) do |value|
     @backgrounds = {
```

Comment:
Let's put `@backgrounds` initialization inside `after_initialize`

```suggestion
  def after_initialize
    @backgrounds = {
```